WooCommerce/WooCommerceTests/ViewRelated/Products/Actions Factory/ProductFormActionsFactory+VisibilityTests.swift

```diff
@@ -166,6 +166,35 @@ final class ProductFormActionsFactory_VisibilityTests: XCTestCase {
         // Assert
         XCTAssertFalse(factory.settingsSectionActions().contains(.downloadableFiles(editable: true)))
     }
+
+    // MARK: - Custom fields
+
+    func test_givenExistingProductAndCustomFields_whenCreatingActions_thenCustomFieldsRowIsVisible() {
+        // Arrange
+        let model = EditableProductModel(product: Fixtures.productWithCustomFields)
+
+        // Act
+        let featureFlagService = MockFeatureFlagService(viewEditCustomFieldsInProductsAndOrders: true)
+        let factory = ProductFormActionsFactory(product: model, formType: .edit, featureFlagService: featureFlagService)
```

Comment:
Suggestion: how about creating a for loop with products of different types to confirm that custom fields are available for all product types?

Please also add tests for readonly mode.

Reply:
Good idea about the loop. in https://github.com/woocommerce/woocommerce-ios/pull/14068/commits/a8639e49323860184e3f2c5ab1b1f517a6500632

Reply:
unit tests for readonly mode added in [3b45405](https://github.com/woocommerce/woocommerce-ios/pull/14068/commits/3b45405c1055ab530e5b15b242555ea192e44f83)